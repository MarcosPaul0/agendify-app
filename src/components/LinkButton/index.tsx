import { Link } from 'expo-router';
import { BUTTON_VARIANTS } from '@components/Button/types/buttonVariants.type';
import { ILinkButtonProps } from './interfaces/linkButtonProps.interface';

const BUTTON_VARIANT = {
  [BUTTON_VARIANTS.filled]: 'bg-BLUE_500',
  [BUTTON_VARIANTS.outlined]: 'bg-transparent',
};

const TEXT_VARIANT = {
  [BUTTON_VARIANTS.filled]: 'text-GRAY_50',
  [BUTTON_VARIANTS.outlined]: 'text-BLUE_500',
};

export function LinkButton({
  text,
  variant = 'filled',
  ...rest
}: ILinkButtonProps) {
  return (
    <Link
      className={`
        w-full h-12 justify-center mt-3 border-2 pt-2.5
        border-BLUE_500 rounded-xl ${BUTTON_VARIANT[variant]}
        font-bold text-center text-base ${TEXT_VARIANT[variant]}
      `}
      {...rest}
    >
      {text}
    </Link>
  );
}
