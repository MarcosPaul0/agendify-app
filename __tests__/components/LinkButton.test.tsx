import { render } from '@testing-library/react-native';
import { LinkButton } from '@components/LinkButton';

describe('Link Button', () => {
  it('Match with snapshot', async () => {
    const tree = render(<LinkButton href="fake-route" text="fake=text" />);

    expect(tree).toMatchSnapshot();
  });
});
