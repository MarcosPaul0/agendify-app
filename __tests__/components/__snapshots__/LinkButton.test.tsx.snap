// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Button Match with snapshot 1`] = `
<Text
  accessibilityRole="link"
  href="fake-route"
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#3763A4",
      "borderBottomColor": "#3763A4",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderBottomWidth": 2,
      "borderLeftColor": "#3763A4",
      "borderLeftWidth": 2,
      "borderRightColor": "#3763A4",
      "borderRightWidth": 2,
      "borderTopColor": "#3763A4",
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "borderTopWidth": 2,
      "color": "#F8F9FA",
      "fontSize": 16,
      "fontWeight": "700",
      "height": 48,
      "justifyContent": "center",
      "lineHeight": 24,
      "marginTop": 12,
      "paddingTop": 10,
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  fake=text
</Text>
`;
